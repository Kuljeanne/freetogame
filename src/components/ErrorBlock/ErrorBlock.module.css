.container {
  background-image: url('../../assets/img/error.png');
  background-position: center;
  background-repeat: no-repeat;
  height: 70vh;
  display: flex;
  justify-content: center;
}
.message {
  color: #fff;
  display: inline-block;
  padding: 2rem;
  border-radius: 1rem;
  font-size: 3rem;
  background-color: rgba(30, 59, 100, 0.662);;
}