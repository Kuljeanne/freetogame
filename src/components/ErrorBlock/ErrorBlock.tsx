import React from "react";

import styles from "./ErrorBlock.module.css";
type ErrorBlockType = {
  message?: string;
};
export const ErrorBlock = ({
  message = "Something went wrong",
}: ErrorBlockType) => {
  return (
    <div className={styles.container} data-testid='error-block'>
      <p className={styles.message}>
        <span>{message}</span>, please try again later...
      </p>
    </div>
  );
};
