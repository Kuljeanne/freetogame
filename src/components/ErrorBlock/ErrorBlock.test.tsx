import { render, screen } from "@testing-library/react";

import { ErrorBlock } from "./ErrorBlock";

test("render Error block", () => {
  render(<ErrorBlock />);
  const p = screen.getByText(/please try again later/i);
  const block = screen.getByTestId("error-block");
  expect(p).toBeInTheDocument();
  expect(block).toMatchSnapshot();
});
