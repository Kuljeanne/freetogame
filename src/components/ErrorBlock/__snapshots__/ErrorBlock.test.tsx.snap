// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Error block 1`] = `
<div
  class="container"
  data-testid="error-block"
>
  <p
    class="message"
  >
    <span>
      Something went wrong
    </span>
    , please try again later...
  </p>
</div>
`;
